python/pyarrow/array.pyx

```diff
@@ -335,8 +335,15 @@ def from_pandas_series(object series, object mask=None, timestamps_to_ms=False):
 
     timestamps_to_ms : bool, optional
         Convert datetime columns to ms resolution. This is needed for
-        compability with other functionality like Parquet I/O which
+        compatibility with other functionality like Parquet I/O which
         only supports milliseconds.
+    Notes
+    -----
+    Timestamps in Arrow are internally stored in UTC, while it's 'viewed'
+    in different timezones when necessary - the underlying data
+    remains immutable.  Similarly, numpy currently does not support
+    timezones either and therefore the .values accessor used by the
+    function series_as_ndarray() performs the implicit conversion to UTC.
```

Comment:
Can you revise to say "Localized timestamps will currently be returned as UTC (pandas's native representation). Timezone-naive data will be implicitly interpreted as UTC".